### PR TITLE
initial draft of versioning guidelines

### PR DESCRIPTION
**Pull request by ghinks**
Issue [158 versioning](https://github.com/nodejs/package-maintenance/issues/158) discourse has resulted in this brief advice for versioning. We are separating the versioning advice from the associated tooling that would or could enforce versioning.

### CODE REVIEW

File: docs/drafts/versioning.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Versioning 
+
+The package maintenance group recommends that supported packages follow the issued NPM [best practices](https://docs.npmjs.com/about-semantic-versioning). Most packages that receive help are expected to already be following this guidance. However if a package is not already on a major semver release number we suggest that bumping up to version 1.0.0 on the next semver major release.
+
```
Comment (Eomm) [inline docs/drafts/versioning.md]: It happens to me these days: I want to understand the new features and breaking changes from a release to another. Very few packages write in the release tag the changes, so could we suggest that is a good practice?

File: docs/drafts/versioning.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+It is also recommended that increasing the minimum required version of node is a breaking change and should be accompanied by major semver change.
```
Comment (vweevers) [inline docs/drafts/versioning.md]: IMO it should be more than a recommendation, it's a must. Bumping major takes no effort and prevents major pains.

File: docs/drafts/versioning.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Versioning 
+
+The package maintenance group recommends that supported packages follow the issued NPM [best practices](https://docs.npmjs.com/about-semantic-versioning). Most packages that receive help are expected to already be following this guidance. However if a package is not already on a major semver release number we suggest that bumping up to version 1.0.0 on the next semver major release.
```
Comment (ljharb) [inline docs/drafts/versioning.md]: npm is always lowercase:
```suggestion
The package maintenance group recommends that supported packages follow the issued npm [best practices](https://docs.npmjs.com/about-semantic-versioning). Most packages that receive help are expected to already be following this guidance. However if a package is not already on a major semver release number we suggest that bumping up to version 1.0.0 on the next semver major release.
```

File: docs/drafts/versioning.md (hunk 1 above)
Comment (ghinks) [reply docs/drafts/versioning.md]: this sounds a most excellent practice.

File: docs/drafts/versioning.md (hunk 2 above)
Comment (ghinks) [reply docs/drafts/versioning.md]: The wording will be changed to make it a **required practice**

File: docs/drafts/versioning.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Versioning 
+
+The package maintenance group requires that supported packages follow the issued 
+NPM [best practices](https://docs.npmjs.com/about-semantic-versioning). Most packages
```
Comment (ljharb) [inline docs/drafts/versioning.md]:
```suggestion
npm [best practices](https://docs.npmjs.com/about-semantic-versioning). Most packages 
```

File: docs/drafts/versioning.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# Versioning 
+
+The package maintenance group requires that supported packages follow the issued 
+NPM [best practices](https://docs.npmjs.com/about-semantic-versioning). Most packages 
+that receive help are expected to already be following this guidance. However if a 
+package is not already on a major semver release number we insist on bumping up
```
Comment (mcollina) [inline docs/drafts/versioning.md]:
```suggestion
package is not already on a major semver release number we recommend on bumping up 
```

File: docs/drafts/versioning.md (hunk 1 above)
Comment (mcollina) [reply docs/drafts/versioning.md]: Some form of changelog is a good practice. Some projects write it in a file in the repo. Other maintain it in github release tags. Maintaining one is a good practice.

Comment (vweevers) [reply docs/drafts/versioning.md]: There's a benefit to (also) maintaining a changelog in GitHub releases: integration. For example, Greenkeeper will include the release notes in its PRs. Greenkeeper doesn't have to parse files to achieve that, as the changelog is already structured. But, can such a recommendation be made without favoring GitHub? Are similar integrations available for BitBucket and others?

Comment (ghinks) [reply docs/drafts/versioning.md]: The views on the change log from @mcollina and the use of tags from @Eomm have been added.

File: docs/drafts/versioning.md [hunk 6]
```diff
@@ -6,5 +6,9 @@ that receive help are expected to already be following this guidance. However if
 package is not already on a major semver release number we recommend on bumping up 
 to version 1.0.0 on the next semver major release.
 
-It is also recommended that increasing the minimum required version of node is a 
+## Other versioning requirements
+- Increasing the minimum required version of node is a 
 breaking change and should be accompanied by major semver change.
+- Release tag comments
```
Comment (ljharb) [inline docs/drafts/versioning.md]: what does this mean? Certainly there should be a semver tag on each published version - I’m not sure if this is what’s meant here.

Comment (ghinks) [reply docs/drafts/versioning.md]: the originator wanted comments on the tag.

Comment (mcollina) [reply docs/drafts/versioning.md]: I think it’s probably better called “release notes”.

Comment (ljharb) [reply docs/drafts/versioning.md]: That would then be:
```suggestions
- release notes, in at least one of or possibly all of: in the commit message that bumps the version; in a changelog file in the version bump commit; in the tag annotation for the semver-tagged published commit.
```

Comment (ghinks) [reply docs/drafts/versioning.md]: k, most reasonable.

**Pull request merged**